arrays.js

```diff
@@ -0,0 +1,276 @@
+    answer = true;
+  }
+  logger(["input Array", givenData, "lastOrFirst", answer]);
+}
+
+// Write a JavaScript program to reverse the elements of a given array of integers of length 3.
+function reverseArray() {
+  const array = [1, 2, 4];
+  let answer;
+  answer = array.reverse();
+  logger(["input Array", array, "reverseArray", answer]);
+}
+
+// Write a JavaScript program to create an array by taking the first and last elements from a given array of integers. The length must be larger than or equal to 1.
+function createByFirstAndLast() {
+  const array = [1, 2, 4];
+  let answer;
+  if (array.length >= 1) {
+    answer = [array[0], array[array.length - 1]];
+  }
+  logger(["input Array", array, "createByFirstAndLast", answer]);
+}
+
+// Write a JavaScript program to find the longest string from a given array of strings.
+function findLongestString() {
+  const givenStrings = ["abcdefghijk", "abcde", "abcdefghij"];
+  let stringLength = 0;
+  let stringName = "";
+  givenStrings.forEach((item) => {
+    if (stringLength < item.length) {
+      stringLength = item.length;
+      stringName = item;
+    }
+  });
+  logger([
+    "input Array",
+    givenStrings,
+    "findLongestString",
+    `The Longest String is ${stringName} which has length of ${stringLength}cm`,
+  ]);
+}
+
+// Write a JavaScript program to divide a given array of positive integers into two parts. First element belongs to the first part, second element belongs to the second part, and third element belongs to the first part and so on. Now compute the sum of two parts and store it in an array of size two.
+function divideAndSum() {
+  const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+  let sum1 = 0;
+  let sum2 = 0;
+  for (let i = 0; i < arr.length; i++) {
+    if (i % 2 === 0) {
+      sum1 += arr[i];
+    } else {
+      sum2 += arr[i];
+    }
+  }
+  logger([["input Array", array, "divideAndSum", sum1, sum2]]);
+}
+
+// Write a JavaScript program to determine if two arrays of integers of the same length are similar. The arrays will be similar if one array can be obtained from another array by swapping at most one pair of elements
+function areArraysSimilar(arr1, arr2) {
+  if (arr1.length !== arr2.length) {
+    // Check both parameter length
+    return false;
+  }
+
+  let diference = []; // Find Differece in Array
+  for (let i = 0; i < arr1.length; i++) {
+    if (arr1[i] !== arr2[i]) {
+      diference.push(i);
+    }
+  }
+
+  if (diference.length === 0) {
+    // No Difference found then true
+    return true;
+  } else if (diference.length === 2) {
+    [i, j] = diference;
+    return arr1[i] === arr2[j] && arr1[j] === arr2[i];
+  } else {
+    return false;
+  }
+}
+
+function areSimilar() {
+  // This is the Main Function which then call areArraysSimilar()
+  const array1 = [1, 2, 3]; // Array 1
+  const array2 = [1, 3, 2]; // Array 2
+  logger(["input Array", array1, array1, areArraysSimilar(array1, array2)]); // true
```

Comment:
We should change the name of the variable.

Reply:
Yes, I changed that on pull request 3, That is Yesterday